tests/pretty/delegation-inherit-attributes.rs

```diff
@@ -52,5 +52,49 @@ reuse to_reuse_functions::const_fn_extern;
 #[must_use = "some reason"]
 reuse to_reuse_functions::async_fn_extern;
 
+mod recursive {
+    // Check that `baz` inherit attribute from `foo`
+    mod first {
+        fn bar() {}
+        #[must_use = "some reason"]
+        reuse bar as foo;
+        reuse foo as baz;
+    }
+
+    // Check that `baz` don't inherit attribute from `bar`
+    mod second {
+        #[must_use = "some reason"]
+        fn bar() {}
+
+        reuse bar as foo;
+        reuse foo as baz;
```

Comment:
I'd expect `baz` to inherit the attribute, because `baz` inherits it from `foo`, which inherits it from `bar`.